### ISSUE TITLE
set process.env.NODE_ENV in node side code poi.config.js

### ISSUE THREAD

**Issue by magicdawn** [egoist/poi#176]
**Do you want to request a *feature* or report a *bug*?**

the case is I need webpack devtool config `inline-source-map` in development for debug case and leave poi(this module) handle the production environment

`poi build` does not set `process.env.NODE_ENV` to `production` for node side code `poi.config.js`, in `poi build`, `NODE_ENV` is `development`

**What is the current behavior?**

- `poi build`
- `process.env.NODE_ENV === 'development'` // true

**If the current behavior is a bug, please provide the steps to reproduce.**
...

**What is the expected behavior?**

- `poi build`
- `process.env.NODE_ENV === 'production'` // true

**If this is a feature request, what is motivation or use case for changing the behavior?**

Guess It's nicer to set this, or users have to explicit set NODE_ENV to change webpack config base on `NODE_ENV`.

If there got other considerations, I'm glad to hear. :)

**Comment by egoist** [comment]
Just released `v8.1.1` which has `process.env.NODE_ENV` set if it's undefined.

```bash
poi // development
poi build // production
poi test // test
poi watch // development
```

BTW, You can use `options.mode` as well:

```js
module.exports = options => ({
  // mode could be `development` `test` `production` `watch`
  // if sourceMap is neither a string or `false`, we will use the default value
  sourceMap: options.mode === 'development' ? 'inline-source-map' : true
})
```

**Comment by magicdawn** [comment]
Good Job 👍
